lib/libhoney/transmission.rb

```
@@ -45,42 +44,45 @@ def add(event)
     def send_loop
       http_clients = Hash.new do |h, api_host|
         h[api_host] = HTTP.persistent(api_host).headers(
-            'User-Agent' => @user_agent,
-            'Content-Type' => 'application/json',
+          'User-Agent'   => @user_agent,
+          'Content-Type' => 'application/json'
         )
       end
 
       # eat events until we run out
-      loop {
-        e = @send_queue.pop
-        break if e == nil
+      loop do
+        event = @send_queue.pop
+        break if event.nil?
 
         before = Time.now
 
         begin
-          http = http_clients[e.api_host]
+          http = http_clients[event.api_host]
+          url  = '/1/events/' + Addressable::URI.escape(event.dataset.dup)
 
-          resp = http.post('/1/events/'+URI.escape(e.dataset), json: e.data, headers: {
-            'X-Honeycomb-Team' => e.writekey,
-            'X-Honeycomb-SampleRate' => e.sample_rate,
-            'X-Event-Time' => e.timestamp.iso8601(3)
-          })
+          resp = http.post(url,
+                           json: event.data,
+                           headers: {
+                             'X-Honeycomb-Team'       => event.writekey,
+                             'X-Honeycomb-SampleRate' => event.sample_rate,
+                             'X-Event-Time'           => event.timestamp.iso8601(3)
+                           })
 
           # "You must consume response before sending next request via persistent connection"
           # https://github.com/httprb/http/wiki/Persistent-Connections-%28keep-alive%29#note-using-persistent-requests-correctly
           resp.flush
 
-          response = Response.new(:status_code => resp.status)
-        rescue Exception => error
+          response = Response.new(status_code: resp.status)
+        rescue StandardError => error
```

I know `rescue Exception` is generally bad practice, but here we really do want to catch everything. As the comment immediately below says, an uncaught exception here will kill one of the threads in the thread pool, and (prior to Ruby 2.5) won't even output anything to say that it happened.

Since #30 the thread pool will at least replenish itself, so losing threads isn't as bad as it used to be, but this still means at minimum a silently dropped event.